python/_dynet.pyx

```diff
@@ -1143,28 +1143,70 @@ cdef class ParameterCollection: # {{{
         cdef LookupParameters pp = LookupParameters.wrap_ptr(p)
         return pp
 
-    cpdef add_parameters(self, dim, PyInitializer init=None, name="", device=""):
-        """Add a parameter to the ParameterCollection
+
+    cpdef add_parameters(self, dim, init=None, name="", device="", scale=1.0, mean=0.0, std=1.0):
+        """Add a parameter to the ParameterCollection with a given initializer. There are different ways of specifying an initializer:
+
+        .. code-block:: python
+            
+            p = m.add_parameters((3,5), init=0)                         # Creates 3x5 matrix filled with 0 (or any other float)
+            p = m.add_parameters((3,5), init='uniform', scale=a)        # Creates 3x5 matrix initialized with U([-a,a])
+            p = m.add_parameters((3,5), init='normal', mean=a, std=b)   # Creates 3x5 matrix initialized with N(a, b)
+            p = m.add_parameters((5,5), init='identity')                # Creates 5x5 identity matrix
+            p = m.add_parameters((5,5), init='saxe')                    # Creates 5x5 orthogonal matrix (NOT SUPPORTED YET)
+            p = m.add_parameters((3,5), init='glorot')                  # Creates 3x5 matrix with glorot init
+            p = m.add_parameters((3,5), init='he')                      # Creates 3x5 matrix with he init
+            p = m.add_parameters(np.zeros((3,5)))                       # Creates 3x5 matrix from a numpy array (size is inferred)
+            p = m.add_parameters((3,5), init=dy.PyInitializer())        # Any parameter initializer
         
         Args:
-            dim (tuple): Shape of the parameter
+            dim (tuple, np.ndarray): Shape of the parameter. If only a numpy array is passed the dimension is inferred
```

Comment:
I think it is a little confusing that we specifially handle when it is np.ndarray for the `dim`. Users can use NumpyInitializer or `parameters_from_numpy`. I think dim can also be specified by np.array([2,3]).

Reply:
I agree that the name is confusing. However there is no easy way to overload functions and I would really like to make it an option to just specify a numpy array and let the dimension be inferred. Using NumpyInitializer is not an option for LookupParameters because of reshaping.

Reply:
```python
model.add_parameters(arr.shape, init=arr)
```

I think it is better that we keep a **consistent interface**. If we reuse dim, it may be easier to initialize for parameters from numpy array. But I doubt whether it is suitable to make the compromise.